### PR TITLE
Jetpack Search: update setup instructions in README

### PR DESCRIPTION
**Pull request by bluefuton**
The README for Jetpack Search contains out-of-date setup instructions. This PR fixes it.

#### Changes proposed in this Pull Request:

No functional changes.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
No.

#### Proposed changelog entry for your changes:

No changelog required.

### CODE REVIEW

File: modules/search/instant-search/README.md [hunk 1]
```diff
@@ -31,22 +31,10 @@ This also works on WP.com
...
+2. Select a theme of your choice and add a Jetpack Search widget to the site via the Customizer. If using a theme with a sidebar widget area, please add the Jetpack Search widget there.
 
-3. Select a theme of your choice and add a Jetpack Search widget to the site via the customizer. If using a theme with a sidebar widget area, please add the Jetpack Search widget there.
+3. Add a Jetpack Search Widget to the Jetpack Search Sidebar and configure the filters you'd like to show in the search overlay.
 
-4. Add a Jetpack Search Widget to the Jetpack Search Sidebar and configure the filters you'd like to show in the search overlay.
-
-5. Navigate to your site's search page (e.g. `/?s=hello`) via the widget you added in (3).
-
-## Architectural Choices
```
Comment (bluefuton) [inline modules/search/instant-search/README.md:44]: The architectural choices are interesting background, but probably not essential for a README.

File: modules/search/instant-search/README.md [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+3. Modify/improve the code in the instant-search directories. New packages should be added to the common package.json.
 4. Run `yarn build-search [--watch]` to compile your changes.
-5. Now test your changes on the front end of your test site.
+5. Test your changes on the front-end of your test site.
 6. Open a PR, and a WordPress.com diff will be automatically generated with your changes.
-7. Test the WordPress.com diff
+7. Test the WordPress.com diff.
 8. Once the code works well in both environments and has been approved by a Jetpack crew member, you can merge your branch!
 
 ### Basic Build Process
```
Comment (bluefuton) [inline modules/search/instant-search/README.md:20]: @jsnmoon do we still need to specify these yarn and node versions? (I'm using yarn v1.22 and node v12.14 locally and all seems to build fine.)

Comment (bluefuton) [inline modules/search/instant-search/README.md:20]: Once #15748 lands, we could just go with `yarn watch` rather than the two `build` steps.

Comment (jsnmoon) [reply modules/search/instant-search/README.md:20]: I don't think it's necessary anymore.

**Review by jeherve — APPROVED** [review]

**Review by jsnmoon — APPROVED** [review]

**Pull request merged**